cassiopeia/solver/SpectralNeighborJoiningSolver.py

```diff
@@ -0,0 +1,281 @@
+"""
+This file stores a subclass of DistanceSolver, SpectralNeighborJoining. This algorithm is based on the one developed by Jaffe, et al. (2020) in their paper titled "Spectral neighbor joining for reconstruction of latent tree models. 
+"""
+
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import abc
+import networkx as nx
+import numpy as np
+import pandas as pd
+from itertools import chain
+
+# Only for debugging purposes
+# if True:
+#     import sys
+#     sys.path.insert(0, '../..')
+
+from cassiopeia.solver.DistanceSolver import DistanceSolver, DistanceSolverError
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.solver import (
+    dissimilarity_functions,
+    solver_utilities,
+)
+
+from sklearn.decomposition import TruncatedSVD
+from scipy.sparse.linalg import svds, eigs
+from scipy.linalg import svd as scipy_svd
+
+
+class SpectralNeighborJoiningSolver(DistanceSolver):
+    """
+    Spectral Neighbor-joining class for Cassiopeia.
+
+    Implements a variation on the Spectral Neighbor-Joining algorithm described by Jaffe et al. in their 2020 paper. This class implements the abstract class DistanceSolver. This class inherits the 'solve' method from DistanceSolver with the find_cherry method implemented using a spectral method as in Jaffe et al. (2020).
+
+    Args:
+        dissimilarity_function: A function by which to compute the dissimilarity map, corresponding to the affinity R(i, j) in Jaffe et al. (2020). By default we will use exp(-d(i,j)) where d is the metric given by weighted_hamming_distance from dissimilarity functions without weights given.
```

Comment:
Be sure all lines (including comments) are 80 characters maximum.

Also, you'll have to add more arguments beyond just dissimilarity function. Take a look at the formatting of other `Args` documentation in the other solvers.